pkg/webservice/handlers.go

```diff
@@ -619,6 +625,57 @@ func getQueueApplications(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getPartitionApplicationIDsByState(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+	vars := httprouter.ParamsFromContext(r.Context())
+	if vars == nil {
+		buildJSONErrorResponse(w, MissingParamsName, http.StatusBadRequest)
+		return
+	}
+	partition := vars.ByName("partition")
+	appState := vars.ByName("state")
+
+	partitionContext := schedulerContext.GetPartitionWithoutClusterID(partition)
+	if partitionContext == nil {
+		buildJSONErrorResponse(w, PartitionDoesNotExists, http.StatusNotFound)
+		return
+	}
+	if !allowedAppStates[appState] {
+		buildJSONErrorResponse(w, "Application state is not allowed: "+appState, http.StatusBadRequest)
+		return
+	}
+	appIDs := partitionContext.GetAppIDsByState(appState)
```

Comment:
I've just realized this... A big question here: if we already do state-based filtering here, do we need to do that again in `GetAppIDsByState()`? We shouldn't have the same logic in two places, it's not needed. What we expose on REST is an interface decision, so I'd rather have it here only. `getAppsByState()` should have an exposed pair `GetAppsByState()` which we call here.

Reply:
For the Rejected applications, they were stored in rejectedApplications map. And the get function is getRejectedAppsByState(). Same for completed application.
Should we place the judge logic in handlers or partition?

Reply:
hi @pbacsko , If i understand correctly, you mean create exposed pair method for getAppsByState(), getRejectedAppsByState() & getCompletedAppsByState(). And let restful handler decide which function to call.
plz correct me if i'm wrong.

Reply:
Yes that's the idea, something like that. Just avoid the double filtering which is error prone. Make a decision at one place and the right place is here, the handler method.